### PR TITLE
TEG resprite

### PR DESCRIPTION
**Pull request by alsoandanswer**
Ported from Bay.

dreamseeker_njPoLshIAg

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 1]
```diff
@@ -76,15 +77,19 @@
...
 /obj/machinery/atmospherics/binary/circulator/update_icon()
-	if(stat & (BROKEN|NOPOWER) || !anchored)
-		icon_state = "circ-p"
-	else if(last_pressure_delta > 0 && recent_moles_transferred > 0)
-		if(last_pressure_delta > 5*ONE_ATMOSPHERE)
-			icon_state = "circ-run"
+	icon_state = anchored ? "circ-assembled" : "circ-unassembled"
+	cut_overlays()
+	if (stat & (BROKEN|NOPOWER) || !anchored)
+		return 1
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:83]:
```suggestion
		return TRUE
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 2]
```diff
@@ -76,15 +77,19 @@
...
+	if (last_pressure_delta > 0 && recent_moles_transferred > 0)
+		if (temperature_overlay)
+			overlays += image('icons/obj/power.dmi', temperature_overlay)
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:86]:
```suggestion
			add_overlay(temperature_overlay)
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 3]
```diff
@@ -76,15 +77,19 @@
...
+		if (last_pressure_delta > 5*ONE_ATMOSPHERE)
+			overlays += image('icons/obj/power.dmi', "circ-run")
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:88]:
```suggestion
			add_overlay("circ-run")
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 4]
```diff
@@ -76,15 +77,19 @@
...
 		else
-			icon_state = "circ-slow"
+			overlays += image('icons/obj/power.dmi', "circ-slow")
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:90]:
```suggestion
			add_overlay("circ-slow")
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 5]
```diff
@@ -76,15 +77,19 @@
...
 	else
-		icon_state = "circ-off"
+		overlays += image('icons/obj/power.dmi', "circ-off")
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:92]:
```suggestion
		add_overlay("circ-off")
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 6]
```diff
@@ -76,15 +77,19 @@
...
 
 	return 1
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:94]:
```suggestion
	return TRUE
```

File: code/modules/atmospherics/components/binary_devices/circulator.dm [hunk 7]
```diff
@@ -97,6 +102,7 @@
 					"You hear a ratchet")
```
Comment (SleepyGemmy) [inline code/modules/atmospherics/components/binary_devices/circulator.dm:102]:
```suggestion
					"You hear a ratchet.")
```

File: code/modules/power/generator.dm [hunk 8]
```diff
@@ -69,11 +73,29 @@
...
-		add_overlay("teg-op[lastgenlev]")
+	if (circ1)
+		circ1.temperature_overlay = null
+	if (circ2)
+		circ2.temperature_overlay = null
+	if (stat & (NOPOWER|BROKEN))
+		return TRUE
+	else
+		if (lastgenlev != 0)
+			overlays += image('icons/obj/power.dmi', "teg-op[lastgenlev]")
```
Comment (SleepyGemmy) [inline code/modules/power/generator.dm:89]:
```suggestion
			add_overlay("teg-op[lastgenlev]")
```

File: code/modules/power/generator.dm [hunk 9]
```diff
@@ -107,6 +129,7 @@
 			else
 				air2.temperature = air2.temperature + heat/air2_heat_capacity
 				air1.temperature = air1.temperature - energy_transfer/air1_heat_capacity
+			playsound(src.loc, 'sound/effects/beam.ogg', 25, 0, 10, , required_preferences = ASFX_AMBIENCE)
```
Comment (SleepyGemmy) [inline code/modules/power/generator.dm:132]:
```suggestion
			playsound(get_turf(src), 'sound/effects/beam.ogg', 25, FALSE, 10, , required_preferences = ASFX_AMBIENCE)
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]
suggested changes addressed.

File: code/game/machinery/atmoalter/portable_atmospherics.dm [hunk 10]
```diff
@@ -121,6 +121,7 @@
 		if(connected_port)
 			disconnect()
 			to_chat(user, "<span class='notice'>You disconnect \the [src] from the port.</span>")
+			playsound(src.loc, W.usesound, 50, 1)
```
Comment (SleepyGemmy) [inline code/game/machinery/atmoalter/portable_atmospherics.dm:124]:
```suggestion
			playsound(get_turf(src), W.usesound, 50, 1)
```

File: code/game/machinery/atmoalter/portable_atmospherics.dm [hunk 11]
```diff
@@ -129,6 +130,7 @@
 			if(possible_port)
 				if(connect(possible_port))
 					to_chat(user, "<span class='notice'>You connect \the [src] to the port.</span>")
+					playsound(src.loc, W.usesound, 50, 1)
```
Comment (SleepyGemmy) [inline code/game/machinery/atmoalter/portable_atmospherics.dm:133]:
```suggestion
					playsound(get_turf(src), W.usesound, 50, 1)
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]
suggested changes addressed.

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**